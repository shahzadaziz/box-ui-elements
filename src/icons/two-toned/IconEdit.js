// @flow
import * as React from 'react';

import AccessibleSVG from '../accessible-svg';
import type { TwoTonedIcon } from '../flowTypes';
import { boxBlue, white } from '../../styles/variables';

const IconEdit = ({ className = '', height = 24, title, width = 24 }: TwoTonedIcon) => (
    <AccessibleSVG
        className={`bdl-IconEdit ${className}`}
        height={height}
        title={title}
        viewBox="0 0 32 32"
        width={width}
    >
        <circle className="background-color" fill={boxBlue} cx="16" cy="16" r="16" />
        <g className="foreground-color" transform="translate(9.000000, 9.000000)" fill={white}>
            <path d="M2.55318931,8.10044342 L10.0254604,0.698424385 C10.9120573,-0.210024968 12.3753683,-0.235595915 13.2938572,0.641310075 C14.212346,1.51821607 14.2381995,2.96553255 13.3516027,3.87398194 L5.77538968,11.3331153 L2.55318931,8.10044342 Z M1.4329261,9.36838187 L4.70132289,12.6010538 L0.774627105,13.9375294 C0.174073632,14.1431411 -0.13775221,13.8233007 0.0585825795,13.2407344 L1.4329261,9.36838187 Z" />
        </g>
    </AccessibleSVG>
);

export default IconEdit;
