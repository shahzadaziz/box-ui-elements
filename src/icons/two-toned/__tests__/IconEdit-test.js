import React from 'react';

import IconEdit from '../IconEdit';

describe('icons/general/IconEdit', () => {
    const getWrapper = (props = {}) => shallow(<IconEdit {...props} />);

    test('should correctly render default icon', () => {
        const wrapper = getWrapper();
        expect(wrapper).toMatchSnapshot();
    });

    test('should correctly render icon with specified width and height', () => {
        const width = 16;
        const height = 17;
        const wrapper = shallow(<IconEdit height={height} width={width} />);

        expect(wrapper.find('AccessibleSVG').prop('width')).toEqual(width);
        expect(wrapper.find('AccessibleSVG').prop('height')).toEqual(height);
    });

    test('should correctly render icon with title', () => {
        const title = 'eidt-icon';
        const wrapper = shallow(<IconEdit title={title} />);

        expect(wrapper.find('AccessibleSVG').prop('title')).toEqual(title);
    });
});
