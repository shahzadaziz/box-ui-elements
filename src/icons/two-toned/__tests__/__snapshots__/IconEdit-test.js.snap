// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icons/general/IconEdit should correctly render default icon 1`] = `
<AccessibleSVG
  className="bdl-IconEdit "
  height={24}
  viewBox="0 0 32 32"
  width={24}
>
  <circle
    className="background-color"
    cx="16"
    cy="16"
    fill="#0061d5"
    r="16"
  />
  <g
    className="foreground-color"
    fill="#fff"
    transform="translate(9.000000, 9.000000)"
  >
    <path
      d="M2.55318931,8.10044342 L10.0254604,0.698424385 C10.9120573,-0.210024968 12.3753683,-0.235595915 13.2938572,0.641310075 C14.212346,1.51821607 14.2381995,2.96553255 13.3516027,3.87398194 L5.77538968,11.3331153 L2.55318931,8.10044342 Z M1.4329261,9.36838187 L4.70132289,12.6010538 L0.774627105,13.9375294 C0.174073632,14.1431411 -0.13775221,13.8233007 0.0585825795,13.2407344 L1.4329261,9.36838187 Z"
    />
  </g>
</AccessibleSVG>
`;
